### PR TITLE
Signup and Roles Page with Specs

### PR DESCRIPTION
**Pull request by AkivaGreen**
Addresses #15 
Addresses #17

### CODE REVIEW

File: app/assets/stylesheets/application.css [hunk 1]
```diff
@@ -26,3 +26,11 @@ p.errors{
   border-radius: 6px;
   text-align: center;
 }
+
+.btn-primary.delete{
+  background-color: #b73434;
+}
+
+td.roles, th.roles {
+    text-align: center;
```
Comment (dfaulken) [inline app/assets/stylesheets/application.css]: 2 spaces please

File: app/views/admin/users/index.haml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+.users
+  %table.users
+    %tr
+      %th Name
+      %th.roles Admin
```
Comment (dfaulken) [inline app/views/admin/users/index.haml]: Seems like applying CSS based on an identifier for the table would be simpler than attaching the same identifier to nearly every th and td.

**Comment by dfaulken** [comment]
I thought we removed all the devise views that we're not modifying ourselves? Anyway, please do that.

**Comment by dfaulken** [comment]
Travis CI shows some test failures.

**Comment by dfaulken** [comment]
Don't forget to run `rubocop` - CodeClimate has found some legitimate issues.

File: spec/controller/scoreboard_partial_spec.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+require 'rails_helper' do
```
Comment (dfaulken) [inline spec/controller/scoreboard_partial_spec.rb]: Whoops

File: spec/features/roles_spec.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+describe 'assigning roles' do
+  context 'with admin privilege' do
+    it 'assigns one role to user' do
+      admin = create :user, admin: true
+      login_as admin
+      visit admin_users_url
+      # binding.pry
```
Comment (dfaulken) [inline spec/features/roles_spec.rb]: Whoops

File: spec/features/roles_spec.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      check 'user_master_of_ceremonies'
+      click_on 'Save'
+      expect(page).to have_text 'User has been updated.'
+      expect(admin.master_of_ceremonies).to eql true
```
Comment (dfaulken) [inline spec/features/roles_spec.rb]: In a feature test, better to check on a user experience to see that this change has taken effect.

File: spec/controller/scoreboard_partial_spec.rb (hunk 3 above)
Comment (AkivaGreen) [inline spec/controller/scoreboard_partial_spec.rb]: Sorry this is part of a separate branch that we are working on. However, it leaked into our master. I will be sure to delete or just leave it blank. We will get to creating all the controller tests in one future pull request.

File: spec/models/participant_spec.rb [hunk 6]
```diff
@@ -20,27 +20,131 @@
...
+      expected = [participant_1, participant_2, participant_3]
+      expect(Participant.scoreboard_order(:participant_number)).to eq expected
+    end
+  end
+end
+
+describe 'top_20' do
+  context'doesnt include anyone after 20th person' do
+    it 'holds 20 people' do
+      21.times { create :participant }
```
Comment (dfaulken) [inline spec/models/participant_spec.rb]: Still not clear why specifically the last person you create should be excluded.

File: spec/models/participant_spec.rb [hunk 7]
```diff
@@ -20,27 +20,142 @@
...
+end
+
+describe 'top_20' do
+  context 'excluding anyone not in top 20' do
+    it 'exludes participant with 21st highest score' do
+      20.times { create :maneuver_participant, :perfect_score }
+      create :maneuver_participant, :perfect_score,
+             reversed_direction: 2
+      top_20 = Participant.top_20
+      expect !top_20.include?(Participant.last)
```
Comment (dfaulken) [inline spec/models/participant_spec.rb]: expect(top_20).not_to include Participant.last

**Comment by dfaulken** [comment]
Please run `git pull upstream master` on this branch.

File: spec/models/participant_spec.rb [hunk 8]
```diff
@@ -140,11 +140,22 @@
 end
 
 describe 'top_20' do
-  context'doesnt include anyone after 20th person' do
-    it 'holds 20 people' do
-      21.times { create :participant }
-      partip_array = Participant.top_20
-      expect !partip_array.include?(Participant.last)
+  context 'excluding anyone not in top 20' do
+    it 'exludes participant with 21st highest score' do
```
Comment (dfaulken) [inline spec/models/participant_spec.rb]: `excludes`

**Pull request merged**